### DISCUSSION TITLE
Trying to understand Plugins - adding a javascript file to Publii

### DISCUSSION THREAD

**Discussion by bjazmoore** [GetPublii/Publii#1734] [Help] [answered]
It looks like I can add resources and Publii will store them somewhere for me to use from my plugin.  In this case it is a Javascript file that I want to inject some code to use in a page/post.  I can handle the injection stuff.  I am stuck with where my file went.  I put it here:

```
your-plugin-folder/
│
├── front-assets/
│   └── my_javascript_file.js
│
├── LICENSE
├── main.js
├── plugin.json
└── thumbnail.svg
```

Should I be moving that file myself in main.js or does Publii move it somewhere from which I can access and use it?

Thanks

**Comment by bobmitro** [comment, marked as answer]
The front-assets files will land in this location: /media/plugins/NAME-OF-PLUGIN/.

Remember that you need to specify that the file should be handled by Publii and treated as an asset by adding an entry in the plugin's config file (**plugin.json**):

```
"assets": {
   "front": [
      "my_javascript_file.js"
   ]
}
```
Also, in the **main.js** file, you need to reference this path, e.g.:
```
let script = `
   <script src="${rendererInstance.siteConfig.domain}/media/plugins/NAME-OF-PLUGIN/my_javascript_file.js"></script>
`
```

**Reply by bjazmoore** [reply]
Thank you so much Bob.  This was not clear to me - especially the way the file is referenced in main.js.  I really appreciate the help.
